### DISCUSSION TITLE
Support for JavaScript local storage

### DISCUSSION THREAD

**Discussion by srini8080** [readium/kotlin-toolkit#501] [General]
I am not able to use the persistent storage between html files in the same epub. While it is working on the other readers, it is not working on readium.

**Comment by mickael-menu** [comment]
Yes, `localStorage` is not supported at the moment.

**Reply by srini8080** [reply]
sessionstorage?

**Reply by srini8080** [reply]
also any other variable I can use within the epub that is exposed from Reader that can be used to pass data between html pages in an epub? Thanks

**Reply by mickael-menu** [reply]
All persistent storage is disabled in the web views, for safety reasons. It can be implemented safely but requires some work and it's not planned at the moment. Are you able to share with us a book that requires persistent storage to work properly?

**Reply by mickael-menu** [reply]
> also any other variable I can use within the epub that is exposed from Reader that can be used to pass data between html pages in an epub? Thanks

No but that's an interesting use case. A sample EPUB would be helpful here too. Do you absolutely need the storage to be persisted across sessions, or only during one reading session?

**Reply by srini8080** [reply]
Yes I will. A kids book that takes the name of the kid in the 1st page and as the html pages are loaded, changes the name of the protagonist to the name of kid. Thanks

**Reply by srini8080** [reply]
And good for one reading session.

**Comment by lavrentius-dev** [comment]
Any news/progress on this? I need to implement some  exercises through an EPUB document that will include instructions in customized tab content and track user progress by saving the state of radio buttons/checkboxes. I consider this a better option than developing a standalone application, which would require periodic updates. I think that as long as the EPUB cannot access external links, using localStorage should be safe. Thanks.

**Reply by mickael-menu** [reply]
Agreed, this would be a great fit. We didn't implement it yet but we're thinking about it maybe this year. I can't give an ETA though.

**Reply by mickael-menu** [reply]
And of course, we welcome any contributions if it's something critical enough for you to start working on it in the toolkit.
